src/System.Private.Uri/tests/FunctionalTests/UriTests.cs

```diff
@@ -896,5 +896,14 @@ public static void TestCasingWhenCombiningAbsoluteAndRelativeUris()
             Uri u = new Uri(new Uri("http://example.com/", UriKind.Absolute), new Uri("C(B:G", UriKind.Relative));
             Assert.Equal("http://example.com/C(B:G", u.ToString());
         }
+
+        [Fact]
+        public static void Uri_ColonInLongRelativeUri_SchemeSuccessfullyParsed()
+        {
+            Uri absolutePart = new Uri("http://www.contoso.com");
+            string relativePart = "a/" + new String('a', 1024) + ":";
```

Comment:
We can define a `const` variable (i.e., `MaxUriSchemeName` to explain the purpose) and use it here, instead of a number.

Reply:
After talking to @caesar1995 about this in person I'm going to add a comment here describing why 1024 is significant.